src/Numerics/DGMethods/DGModel.jl

```diff
@@ -719,6 +762,15 @@ function (dg::DGModel)(tendency, state_prognostic, _, t, α, β)
     # The synchronization here through a device event prevents CuArray based and
     # other default stream kernels from launching before the work scheduled in
     # this function is finished.
+    if dg.tendency_filter !== nothing
+        comp_stream = Filters.apply!(
+            tendency,
+            1:num_state_tendency,
+            dg.grid,
+            dg.tendency_filter;
+            dependencies = comp_stream
+        )
+    end
```

Comment:
I'm trying to debug this. 
I tested this with `experiments/TestCase/isothermal_zonal_flow.jl` and here `comp_stream = nothing` and the code breaks at the following line, with error:
```
ERROR: LoadError: MethodError: no method matching wait(::KernelAbstractions.CPU, ::Nothing)
```
I'm trying to understand why in this case the `apply_async!` returns nothing to the `Filters.apply!`. @simonbyrne , @jkozdon any clue?

Reply:
Ah! I rolled back to commit [ac05980](https://github.com/CliMA/ClimateMachine.jl/pull/1953/commits/ac059805ed895efb7e08bf2e7e453a6f1c1094d1), that was before applying @jkozdon 's suggestion and it works. Hence, something needs to be improved, building on that suggestion. Thanks :)

Reply:
Update: This has been resolved by @mwarusz 's fixes! Thanks @jkozdon  and @mwarusz !!!